server/lib/manifest.coffee

```diff
@@ -17,22 +42,39 @@ class exports.Manifest
             else
                 providerName = providerName[0]
 
-                # This could be moved to a separate factory class...
                 if providerName is "gitlab.cozycloud.cc"
                     Provider = require('./git_providers').CozyGitlabProvider
-                else # fallback to github
+
+                else
                     Provider = require('./git_providers').GithubProvider
 
                 provider = new Provider app
                 provider.getManifest (err, data) =>
                     @config = {}
                     @config = data unless err?
-                    callback err
+                    callback err, data
         else
             @config = {}
-            logger.warn 'App manifest without git URL'
+            logger.warn(
+                'App manifest without recognized git URL or package field')
             logger.raw app
-            callback null
+            callback null, {}
+
+
+    # Download the manifest from the NPM registry. This manifest has many
+    # information (it includes history). So we extract the latest version
+    # of the application manifest.
+    downloadFromNpm: (packageName, callback) ->
+        client = request.createClient "https://registry.npmjs.org/"
+        client.get packageName, (err, res, data) ->
+            if res.statusCode is 404
```

Comment:
`res?.statusCode`. I think it's possible to have an `err` but no `res` (DNS resolution failure for example).